proposals/0118-partitioned-epoch-reward-distribution.md

```diff
@@ -206,24 +206,29 @@ partition indicated by comparing its current block height to
 comparing a partial sum of the rewards calculation (those partitions expected to
 have been distributed) with the `EpochRewards::distributed_rewards` field.
 
-### Restrict Stake Account Access
+### Restrict Stake Account Mutation
 
-To avoid the complexity and potential rewards in unexpected stake accounts,
-the stake program is disabled during the epoch reward distribution period.
+The stake accounts need to be preserved in credit-only state while distributions
+are in progress in order to prevent any changes interfering with reward
+distribution.
 
-Any transaction that invokes staking program during this period will fail with
-a new unique error code - `StakeProgramUnavailable`.
+Therefore, the Stake Program needs access to a syscall which reports whether the
+distribution phase is active. This new syscall `sol_get_epoch_rewards_status`
+should report the value of `EpochRewards::active`. All Stake Program
```

Comment:
Rather than creating a new syscall, should we use the `sol_get_epoch_rewards_sysvar` syscall and check `active` there?

Reply:
I don't have a strong preference either way. I'm leaning towards keeping the full sysvar in the syscall just for consistency with other sysvars. And it's easy enough to add a function for `get_epoch_rewards_status` which wraps that syscall to pull out the `active` field.

Reply:
Yep, makes sense. I pushed a commit describing the `sol_get_epoch_rewards_sysvar` syscall instead.